### ISSUE TITLE
locale/userLocale settings key can be NULLed under some circumstances

### ISSUE THREAD

**Issue by kannes** [qgis/QGIS#52359]
### What is the bug or the crash?

- Plugin Builder generates code that does assumes `QSettings().value('locale/userLocale')` to return a string.
- `QSettings().value('locale/userLocale')` can return `NULL`/`None` under some circumstances
- Lots of plugins were developed using Plugin Builder and might not be maintained anymore
- Lots of plugins can crash if the user ends up in this situation

References:
https://lists.osgeo.org/pipermail/qgis-user/2023-March/052727.html
https://github.com/g-sherman/Qgis-Plugin-Builder/issues/116

### Steps to reproduce the issue

1. $ qgis --lang de
1. "Einstellungen" -> "Optionen" -> "Allgemein" -> "System-Locale übersteuern" -> "Deutsch" (aka `de`)
1. Quit QGIS
1. $ qgis --lang en
1. "Options" -> "General" -> Uncheck "Override System Locale" (note how its User Interface Translation combobox' current value is empty and grey)

-> `QSettings().value('locale/userLocale')` now returns `NULL`

### Versions

master

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

https://github.com/qgis/QGIS/issues/37620 might be related maybe.

**Comment by bru500** [comment]
This is a nasty error. If in Options - General, 'Override System Local' is selected, very many plugins crash when loaded.
We need then to set locale/userLocal manualy via python console: QSettings().setValue('locale/userLocale', 'en_US')
But after opening Options window and clicking done, again the option is missing and the plugins crash.

**Comment by alexbruy** [comment]
This seems not a QGIS issue, but a problem in Plugin Builder and other plugins which are not checking return value before using it.

**Comment by kannes** [comment]
Yes absolutely, but many plugins have been built with the Plugin Builder and continue to be build with it. Sadly the Plugin Builder is not getting updated but keeps getting used by plugin developers so **this will continue to be an issue**. The issue is not obvious to a developer as it might not trigger for them but happens in "other people's environments".

It would be great to implement a workaround in QGIS as a quicker fix.

Another ticket that had collected broken plugins and thoughts: https://github.com/qgis/QGIS/issues/29281

**Comment by kannes** [comment]
Future reader: Do not use `QSettings`, but use `QgsSettings`.
If you think you are already using `QgsSettings`, check again and make sure.

I got a user where `NULL` is set as value so `QSettings` will *not* use a default value if passed (e.g. `QLocale().name()`:
```
>>> type(QSettings().value("locale/userLocale"))
<class 'PyQt5.QtCore.QVariant'>
>>> QSettings().value("locale/userLocale")
NULL
>>> QSettings().value("locale/userLocale", QLocale().name())
NULL
```

`QgsSettings` however does return a `None` and consider that an unset value, making the default replacement work:
```
>>> type(QgsSettings().value("locale/userLocale"))
<class 'NoneType'>
>>> QgsSettings().value("locale/userLocale")
>>> QgsSettings().value("locale/userLocale", QLocale().name())
'de_DE'
```

The `QGIS3.ini` `locale` section looks like this:
```
[locale]
userLocale=
globalLocale=
overrideFlag=false
showGroupSeparator=false
```

Using `QSettings` works fine if there is no `userLocale=` line in the ini.

-----

This might actually be a bug of `QgsSettings` if it disagress with `QSettings` about what a key without value means?

**Comment by kannes** [comment]
Fixed by @Djedouas's https://github.com/qgis/QGIS/issues/52359 🥳
